src/components/card/CardSidebarTabDetails.vue

```diff
@@ -93,21 +106,31 @@
 				<span class="hidden-visually">{{ t('deck', 'Due date') }}</span>
 			</div>
 			<div class="section-details">
-				<NcDatetimePicker v-model="duedate"
+				<NcDatetimePicker v-if="(!copiedCard.duedate && !card.done) || copiedCard.duedate"
+					v-model="duedate"
 					:placeholder="t('deck', 'Set a due date')"
 					type="datetime"
 					:minute-step="5"
 					:show-second="false"
 					:lang="lang"
 					:formatter="format"
-					:disabled="saving || !canEdit"
+					:disabled="saving || !canEdit || card.done"
 					:shortcuts="shortcuts"
 					confirm />
 				<NcActions v-if="canEdit">
-					<NcActionButton v-if="copiedCard.duedate" icon="icon-delete" @click="removeDue()">
+					<NcActionButton v-if="copiedCard.duedate && !card.done" icon="icon-delete" @click="removeDue()">
```

Comment:
Since this actions menu only has one action, it should simply be an `NcButton` as well.

Reply:
And use a proper material design icon, see the `NcButton` docs.